### DISCUSSION TITLE
Resistive touch display with driver chip XPT2046

### DISCUSSION THREAD

**Discussion by nonix** [arendst/Tasmota#10506] [Ideas]
Hi guys,
I am new to Tasmota, I think it could be useful feature to add support for resistive touch  display such as [LoLin TFT Display](https://www.wemos.cc/en/latest/d1_mini_shield/tft_2_4.html).
The driver chip XPT2046 is connected via SPI, I guess this would require to support multiple SPI devices on the same bus.
Unfortunately, I am not so skilled to add all of that by myself, but sure I can  help.

Thank you for considering,
 N.

**Comment by sfromis** [comment]
If someone creates a Tasmota driver for the display, that vastly increases the chance of support being added.

**Reply by nonix** [reply]
Hi sfromis,

I think the display driver already exists in tasmota 9.x it uses ILI9341. However, the touch (aka digitizer)  controller is different than the (only) one currently supported FT5206. 
This LoLin 2.4" Display uses resistive touch with XPT2046 (SPI) driver chip. The low level driver exists too, it was originally written by Paul Stoffregen [https://github.com/PaulStoffregen/XPT2046_Touchscreen]

I believe the LoLin 2.4" TFT Display  is good product with fair price and would be a great addition to Tasmota family.

I will try to write my "glue" by I am far from an expert as Theo, so if someone more skilled would like to give a hand it would be very much appreciated.

Thank you for considering,
N.

**Reply by barbudor** [reply]
To ease helpers, the best is to have your fork of Tasmota on github with a dedicated branch where you push your changes
This allows to pull your fork and make efficient suggestions to help you.
And at the end you will it to push your PR to Tasmota.

**Reply by barbudor** [reply]
And don't forget to read https://tasmota.github.io/docs/For-Developers/

**Reply by nonix** [reply]
@barbudor , thank you I have followed your advice and have forked, created a branch called "XPT2046" (https://github.com/nonix/Tasmota/tree/XPT2046) Now all changes are committed to this branch and branch was merged with latest Theo's development (as of few minutes ago).
I have to say, the C++ coding is not my daily business but rather hobby, the code provided could be a bit hairy. I would be glad if someone can give me a hand with perhaps a review and suggest the next step.

Thank you,
 N.
